numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -38,6 +38,26 @@
 #include "npy_cblas.h"
 #include "npy_buffer.h"
 
+
+/*
+ * Define a stack allocated dummy array with only the minimum information set:
+ *   1. The descr, the main field interesting here.
+ *   2. The flags, which are needed for alignment;.
+ *   3. The type is set to NULL and the base is the original array, if this
+ *      is used within a subarray getitem to create a new view, the base
+ *      must be walked until the type is not NULL.
+ *
+ * The following should create errors in debug mode (if deallocated
+ * incorrectly), since base would be incorrectly decref'd as well.
+ */
+#define MAKE_DUMMY_STACK_ARRAY(arr, dummy_array)                    \
```

Comment:
Maybe instead
```
static PyArrayObject_fields dummy_stack_array(PyArrayObject *orig) {
    PyArrayObject_fields fields;
    fields.flags = PyArray_FLAGS(orig);
    Py_TYPE(&fields) = NULL;
    fields.base = (PyObject *)orig;
    return fields;
}
```
then at the call site
```
PyArrayObject_fields dummy_fields = dummy_stack_array(arr);
PyArrayObject *dummy_array = (PyArrayObject *)dummy_fields;
```
which is almost certainly optimized to the same code, and feels less hacky